esx_service/vmdk_ops_test.py

```diff
@@ -252,7 +255,156 @@ def test_failure(self):
+        if (vm):
+            logging.debug("Found: VM %s", vm_name)
+            if vm[0].runtime.powerState == vim.VirtualMachinePowerState.poweredOff:
+                logging.debug("Attempting to power on %s", vm_name)
+                task = vm[0].PowerOnVM_Task()
+                vmdk_ops.wait_for_tasks(si, [task])
+    
+    def remove_vm(self, si, vm_name):
+
+        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == vm_name]
+        if (vm):
+            logging.debug("Found: VM %s", vm_name)
+            #logging.debug("The current powerState is  : %s", format(vm[0].runtime.powerState)))
+            if vm[0].runtime.powerState == vim.VirtualMachinePowerState.poweredOn:
+                logging.debug("Attempting to power off %s", vm_name)
+                task = vm[0].PowerOffVM_Task()
+                vmdk_ops.wait_for_tasks(si, [task])
+            
+            logging.debug("Trying to destroy VM %s", vm_name)    
+            task = vm[0].Destroy_Task()
+            vmdk_ops.wait_for_tasks(si, [task])
+
+    def mkdir(self, path):
+        """ Create a directory if it doesn't exist. Returns pathname or None. """
+        if not os.path.isdir(path):
+            try:
+                os.mkdir(path)
+            except OSError as e:
+                return None
+        return path
+
+    def cleanup(self):
+        # remove VM
+        writeable_ds_path = 'datastore1'
+        if writeable_ds_path:
+            # get service_instance, and create a VM
+            if not vmdk_ops.si:
+                vmdk_ops.connectLocal()
+        self.remove_vm(vmdk_ops.si, self.vm_name)
+
+        for v in self.get_testvols():
+            self.assertEqual(
+                None,
+                vmdk_ops.removeVMDK(os.path.join(v['path'], v['filename'])))
+
+    def get_testvols(self):
+        return [x
+                for x in vmdk_utils.get_volumes()
+                if x['filename'].startswith('VmdkAttachDetachTestVol')]
+
+    
+    def testAttachDetach(self):
+        logging.debug("Start VMDKAttachDetachTest")
+
+        if not vmdk_ops.si:
+            vmdk_ops.connectLocal()
+        #find test_vm
+        vm = [d for d in vmdk_ops.si.content.rootFolder.childEntity[0].vmFolder.childEntity if d.config.name == self.vm_name]
+        self.assertNotEqual(None, vm)
+
+        # attach max_vol_count disks
+        for id in range(1, self.max_vol_count+1):
+            volName = 'VmdkAttachDetachTestVol' + str(id)
+            fullpath = os.path.join(path, volName + '.vmdk')
+            ret = vmdk_ops.disk_attach(vmdk_path=fullpath,
+                                       vm=vm[0])
+            self.assertFalse("Error" in ret)
+
+        # attach one more disk, which should fail    
+        volName = 'VmdkAttachDetachTestVol' + str(self.max_vol_count+1)
+        fullpath = os.path.join(path, volName + '.vmdk')
+        ret = vmdk_ops.disk_attach(vmdk_path=fullpath,
```

Comment:
We have four pvscsi controllers max or one LSI controller plus three PVSCSI controllers. For each controller unit #7 is reserved. Leaving 14 slots per controller available for disks Take away one slot for the boot disk of the VM that leaves 55 disks max. Is that correct?